plugins/CoreAdminHome/Model/DuplicateActionRemover.php

```diff
@@ -0,0 +1,193 @@
+
+            $result[] = $dupeInfo;
+        }
+        return $result;
+    }
+
+    /**
+     * Executes one SQL statement that sets all idaction columns in a table to a single value, if the
+     * values of those columns are in the specified set (`$duplicateIdActions`).
+     *
+     * Notes:
+     *
+     * The SQL will look like:
+     *
+     *     UPDATE $table SET
+     *         col1 = IF((col1 IN ($duplicateIdActions)), $realIdAction, col1),
+     *         col2 = IF((col2 IN ($duplicateIdActions)), $realIdAction, col2),
+     *         ...
+     *      WHERE col1 IN ($duplicateIdActions) OR col2 IN ($duplicateIdActions) OR ...
+     *
+     * @param string $table
+     * @param int $realIdAction The idaction to set column values to.
+     * @param int[] $duplicateIdActions The idaction values that should be changed.
+     */
+    public function fixDuplicateActionsInTable($table, $realIdAction, $duplicateIdActions)
+    {
+        $idactionColumns = array_values($this->idactionColumns[$table]);
+        $table = Common::prefixTable($table);
+
+        $inFromIdsExpression = $this->getInFromIdsExpression($duplicateIdActions);
+        $setExpression = "%1\$s = IF(($inFromIdsExpression), $realIdAction, %1\$s)";
+
+        $sql = "UPDATE $table SET\n";
+        foreach ($idactionColumns as $index => $column) {
+            if ($index != 0) {
+                $sql .= ",\n";
+            }
+            $sql .= sprintf($setExpression, $column);
+        }
+        $sql .= $this->getWhereToGetRowsUsingDuplicateActions($idactionColumns, $duplicateIdActions);
+
+        Db::query($sql);
+    }
+
+    /**
+     * Returns the server time and idsite of rows in a log table that reference at least one action
+     * in a set.
+     *
+     * @param string $table
+     * @param int[] $duplicateIdActions
+     * @return array with two elements **idsite** and **server_time**. idsite is the site ID and server_time
+     *               is the date of the log.
+     */
+    public function getSitesAndDatesOfRowsUsingDuplicates($table, $duplicateIdActions)
+    {
+        $idactionColumns = array_values($this->idactionColumns[$table]);
+        $table = Common::prefixTable($table);
+
+        $sql = "SELECT idsite, DATE(server_time) as server_time FROM $table ";
+        $sql .= $this->getWhereToGetRowsUsingDuplicateActions($idactionColumns, $duplicateIdActions);
+        return Db::fetchAll($sql);
+    }
+
+    private function getIdActionTableColumnsFromMetadata()
+    {
+        $result = array();
+        foreach (self::$tablesWithIdActionColumns as $table) {
+            $columns = $this->tableMetadataAccess->getIdActionColumnNames(Common::prefixTable($table));
+
+            $this->logger->debug("Found following idactions in {table}: {columns}", array(
+                'table' => $table,
+                'columns' => implode(',', $columns)
+            ));
+
+            $result[$table] = $columns;
+        }
+        return $result;
+    }
```

Comment:
They can be added via dimension classes (eg, ContentName). I don't know if there are any non-core ones that add idaction columns, but many core ones do so I guess it's conceivable that they exist. Also, I guess it's possible for some plugins to be disabled in which case the columns wouldn't exist.